org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/IdeAttachedProjectConnection.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.internal;
+
+import java.util.function.Supplier;
+
+import org.gradle.tooling.BuildAction;
+import org.gradle.tooling.BuildActionExecuter;
+import org.gradle.tooling.BuildActionExecuter.Builder;
+import org.gradle.tooling.BuildLauncher;
+import org.gradle.tooling.CancellationTokenSource;
+import org.gradle.tooling.GradleConnectionException;
+import org.gradle.tooling.GradleConnector;
+import org.gradle.tooling.LongRunningOperation;
+import org.gradle.tooling.ModelBuilder;
+import org.gradle.tooling.ProjectConnection;
+import org.gradle.tooling.ResultHandler;
+import org.gradle.tooling.TestLauncher;
+import org.gradle.tooling.model.build.BuildEnvironment;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+import org.eclipse.buildship.core.internal.configuration.GradleArguments;
+import org.eclipse.buildship.core.internal.gradle.GradleProgressAttributes;
+
+final class IdeAttachedProjectConnection implements ProjectConnection {
+
+    private final ProjectConnection delegate;
+    private final GradleArguments gradleArguments;
+    private final GradleProgressAttributes progressAttributes;
+
+    private IdeAttachedProjectConnection(ProjectConnection connection, GradleArguments gradleArguments, GradleProgressAttributes progressAttributes) {
+        this.delegate = connection;
+        this.gradleArguments = gradleArguments;
+        this.progressAttributes = progressAttributes;
+    }
+
+    @Override
+    public BuildLauncher newBuild() {
+        return configuratOperation(() -> this.delegate.newBuild());
+    }
+
+    @Override
+    public TestLauncher newTestLauncher() {
+        return configuratOperation(() -> this.delegate.newTestLauncher());
+    }
+
+    @Override
+    public <T> ModelBuilder<T> model(Class<T> modelType) {
+        return configuratOperation(() -> this.delegate.model(modelType));
+    }
+
+    @Override
+    public <T> BuildActionExecuter<T> action(BuildAction<T> buildAction) {
+        return configuratOperation(() -> this.delegate.action(buildAction));
+    }
+
+    private <T extends LongRunningOperation> T configuratOperation(Supplier<T> operationSupplier) {
```

Comment:
:x: The method name has a typo. Moreover, why does it take a `Supplier` only to immediately call `get()` on it? Wouldn't it be easier to just pass a `T` instance?

Reply:
Fixed